### PR TITLE
Marshall/mock urlopen

### PR DESCRIPTION
**Pull request by mpkrass7**
Essentially I just added a couple of web urls to the unit tests and validated that they work. If they aren't valid urls, they won't be tested. Please don't make me make a unit test to assess if the check for url validity is itself valid. 

Notes:
- In one of the unit tests, I don't check the web urls because the unit test checks the name attribute of the resulting colormap. I wasn't sure what the name should be so I left them out of that test
- Hi Kevin

### CODE REVIEW

**Comment by mpkrass7** [comment]
Hmm I'm seeing that some checks are not successful here

**Comment by mpkrass7** [comment]
Yay I think we're passing all of the tests. Adding exception handling for `OSError` was the key

**Review by arvkevi — CHANGES_REQUESTED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -7,6 +7,7 @@ requires = [
     "numpy",
     "pillow",
     "kneed",
+    "requests"
```
[the remark this answers is not in the document]
Comment (mpkrass7) [reply pyproject.toml:10]: Ahh thought I had removed it from there..

**Review by arvkevi — APPROVED** [review]
LGTM! Thank you @mpkrass7, this was a great find!

**Pull request merged**